multigram/README.md

```diff
@@ -0,0 +1,96 @@
+# Multigram -- protocol negotiation and multiplexing over datagrams
+
+For multiplexing different protocols on the same datagram connection, multigram prepends a 1-byte header to every packet. This header represents an index in a table of protocols shared between both endpoints. This protocol table is negotiated by exchanging the intersection of the endpoint's supported protocols. The protocol table's size of 256 tuples can be increased by nesting multiple multigram headers.
```

Comment:
The maximum packet size would be bubbled up from the transports through all the protocol wrappers, e.g. `512 (UDP) - 1 (multigram) - 20 (cryptoauth) - 1 (multigram) - 12 (switch) - 1 (multigram) = 477`